src/igvfd/types/file_set.py

```diff
@@ -274,15 +273,28 @@ def input_for(self, request, input_for):
             },
             'notSubmittable': True
         })
-    def construct_library_sets(self, request, samples=None):
+    def construct_library_sets(self, request, samples=None, input_file_sets=[]):
         construct_library_sets = set()
-        for sample in samples:
-            sample_object = request.embed(sample,
-                                          '@@object_with_select_calculated_properties?'
-                                          'field=construct_library_sets'
-                                          )
-            if sample_object.get('construct_library_sets', []):
-                construct_library_sets = construct_library_sets | set(sample_object.get('construct_library_sets', []))
+        if samples:
+            for sample in samples:
+                sample_object = request.embed(sample,
+                                              '@@object_with_select_calculated_properties?'
+                                              'field=construct_library_sets'
+                                              )
+                if sample_object.get('construct_library_sets', []):
+                    construct_library_sets = construct_library_sets | set(
+                        sample_object.get('construct_library_sets', []))
+        else:
+            all_inputs_cls = True
+            cls_in_input_file_sets = set()
+            if input_file_sets:
+                for input_set in input_file_sets:
+                    if input_set.startswith('/construct-library-sets/'):
+                        cls_in_input_file_sets.add(input_set)
+                    else:
+                        all_inputs_cls = False
+            if all_inputs_cls:
+                construct_library_sets = construct_library_sets | cls_in_input_file_sets
```

Comment:
this line can be simplified: construct_library_sets = cls_in_input_file_sets, but it works fine.